### PR TITLE
ComboBox: implement Sandpack code editors

### PR DESCRIPTION
**Pull request by AlbertCarreras**
ComboBox: implement Sandpack code editors

### CODE REVIEW

File: docs/examples/combobox/controlled.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
+// @flow strict
+import { useState, type Node } from 'react';
+import { Box, ComboBox, Flex, Text } from 'gestalt';
+
+export default function Example(): Node {
+  const US_STATES = [
```
Comment (dangerismycat) [inline docs/examples/combobox/controlled.js:6]: nit: these sort of things shouldn't be defined in components, as they'll be recreated on each render. not important here but a good thing to keep in mind

File: docs/pages/web/combobox.js [hunk 2]
```diff
@@ -6,59 +6,26 @@ import docgen, { type DocGen } from '../../docs-components/docgen.js';
...
-    've / ver',
-    'xe / xem',
-    'xie / xem',
-    'zie / zem',
-  ];
-
-  const options = PRONOUNS.map((pronoun, index) => ({ label: pronoun, value: 'value'+index }));
+    <Page title={generatedDocGen?.displayName}>
+      <PageHeader name={generatedDocGen?.displayName} description={generatedDocGen?.description}>
+        <SandpackExample code={main} name="No image source" hideEditor previewHeight={320} />
```
Comment (dangerismycat) [inline docs/pages/web/combobox.js:26]: nit: I don't understand how this name relates to the code example?

File: docs/pages/web/combobox.js [hunk 3]
```diff
@@ -246,273 +110,23 @@ function ComboBoxExample(props) {
...
-        onClear={resetErrorMessage}
-        options={options}
-        placeholder="Add your pronouns"
-      />
-    </Box>
-  );
-}
-`}
+            sandpackExample={
+              <SandpackExample code={uncontrolled} name="uncontrolled" previewHeight={320} />
```
Comment (dangerismycat) [inline docs/pages/web/combobox.js:114]: nit: `name` is user-facing (it populates the accessibility label on an icon button) so we should use Sentence case

File: docs/pages/web/combobox.js (hunk 2 above)
Comment (dangerismycat) [inline docs/pages/web/combobox.js:26]: nit: if we're using the same height for all the code examples, that should be a top-of-file const

**Review by dangerismycat — APPROVED** [review]

File: docs/examples/combobox/controlled.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/examples/combobox/controlled.js:6]: DONE

File: docs/pages/web/combobox.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/pages/web/combobox.js:26]: Didn't know they were a11y labels.

FIXED DONE

File: docs/pages/web/combobox.js (hunk 3 above)
Comment (AlbertCarreras) [reply docs/pages/web/combobox.js:114]: DONE

File: docs/pages/web/combobox.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/pages/web/combobox.js:26]: DONE

**Pull request merged**